src/clr-angular/forms/select/select.spec.ts

```diff
@@ -0,0 +1,110 @@
+        let ifOpenService: IfOpenService;
+
+        beforeEach(function() {
+            context = this.create(ClrSelect, TestComponent, [IfOpenService]);
+            ifOpenService = context.getClarityProvider(IfOpenService);
+        });
+
+        describe("Typescript API", function() {
+            it("provides a method to toggle the popover on click", () => {
+                expect(ifOpenService.open).toBeUndefined();
+
+                context.clarityDirective.onClick(new MouseEvent("click"));
+
+                expect(open).not.toBe(true);
+
+                context.clarityDirective.onClick(new MouseEvent("click"));
+
+                expect(open).not.toBe(false);
+            });
+
+            it("provides a method to close the popover on focusout", () => {
+                expect(ifOpenService.open).toBeUndefined();
+
+                context.clarityDirective.onFocusOut();
+                expect(ifOpenService.open).toBe(false);
+            });
+        });
+
+        describe("View Basics", () => {
+            it("projects content", () => {
+                expect(context.clarityElement.textContent).toMatch(/Test/);
+            });
+
+            it("adds the .clr-select class on the host", () => {
+                expect(context.clarityElement.classList.contains("clr-select")).toBe(true);
+            });
+
+            it("contains an editable input", () => {
+                const input = context.clarityElement.querySelector(".clr-select-input");
+                expect(input).not.toBeNull();
+                expect(input.hasAttribute("contenteditable")).toBe(true);
+                expect(input.getAttribute("contenteditable")).toBe("true");
+            });
+
+            it("contains a options menu trigger", () => {
+                expect(context.clarityElement.querySelector(".clr-select-trigger")).not.toBeNull();
+            });
+
+            it("opens the menu on the trigger click", () => {
+                const trigger = context.clarityElement.querySelector(".clr-select-trigger");
+
+                expect(ifOpenService.open).toBeUndefined();
+
+                trigger.click();
+
+                expect(ifOpenService.open).toBe(true);
+            });
+
+            it("focuses the input when the trigger is clicked and the menu is open", () => {
+                const trigger = context.clarityElement.querySelector(".clr-select-trigger");
+
+                let focused = context.clarityElement.querySelector(":focus");
+
+                expect(focused).toBeNull();
+
+                trigger.click();
+
+                focused = context.clarityElement.querySelector(":focus");
+
+                expect(focused.classList.contains("clr-select-input")).toBe(true);
+            });
+
+            it("keeps the options menu open when the input is clicked", () => {
+                const trigger = context.clarityElement.querySelector(".clr-select-trigger");
+                trigger.click();
```

Comment:
You're replicating the `"opens the menu on the trigger click"` test here. Just set `ifOpenService.open = true` directly, then test what you really want to test here.